### PR TITLE
feat: switch logo plugin

### PR DESCRIPTION
**Pull request by guanbinrui**
## Description

Closes MF-0000

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/plugins/SwitchLogo/src/SNSAdaptor/SwitchLogoDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    const openApplicationBoardDialog = useOpenApplicationSettings()
+
+    const onSave = useCallback(async () => {
+        if (!identity?.identifier?.userId) return
+        switchLogoSettings[identity.identifier.userId].value = logoType ?? defaultLogoType
+        onClose()
+        if (needShare) {
+            share?.([t.share_text(), t.share_mask()].join('\n'))
+        }
+    }, [logoType, identity?.identifier?.userId, defaultLogoType, share, needShare])
```
Comment (guanbinrui) [inline packages/plugins/SwitchLogo/src/SNSAdaptor/SwitchLogoDialog.tsx:86]:
```suggestion
    }, [logoType, identity?.identifier?.userId, defaultLogoType, share, needShare, onClose])
```

File: packages/app/src/plugin-host/enable.ts [hunk 2]
```diff
@@ -119,6 +119,7 @@ startPluginSNSAdaptor(CurrentSNSNetwork.__SPA__, {
             send: reject,
             NFTAvatarTimelineUpdated: emptyEventRegistry,
             themeSettings: createConstantSubscription(undefined),
+            switchLogoSettings: {},
```
Comment (Jack-Works) [inline packages/app/src/plugin-host/enable.ts:122]: ?

File: packages/mask/shared/legacy-settings/settings.ts [hunk 3]
```diff
@@ -69,3 +69,5 @@ try {
     else if (lng === 'ko') languageSettings.value = LanguageOptions.koKR
     else languageSettings.value = LanguageOptions.__auto__
 } catch {}
+
+export const switchLogoSettings = createBulkSettings('SwitchLogo', SwitchLogoType.New)
```
Comment (Jack-Works) [inline packages/mask/shared/legacy-settings/settings.ts:73]: this should not be stored here.

File: packages/plugin-infra/src/types.ts [hunk 4]
```diff
@@ -343,6 +344,9 @@ export namespace Plugin.SNSAdaptor {
         queryPersonaByProfile: (id: ProfileIdentifier) => Promise<PersonaInformation | undefined>
         connectPersona: () => Promise<void>
         createPersona: () => void
+        getDecentralizedSearchSettings?: () => Promise<boolean>
+        setDecentralizedSearchSettings?: (checked: boolean) => Promise<void>
+        switchLogoSettings: Record<string, ValueRefWithReady<SwitchLogoType>>
```
Comment (Jack-Works) [inline packages/plugin-infra/src/types.ts:349]: this should not be a part of plugin api.

**Pull request merged**